main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
hum, IIUC this just normalizes line endings to `\n`, right? seems wrong to me to do it on MIO's side.

But maybe MIO should consider `\r` a line end though, indeed.  and then also eat a following `\n` probably.
```diff
 				if (s[i] == '\n')
 				{
 					i++;
 					break;
 				}
+				else if (s[i] == '\r')
+				{
+					i++;
+					if (s[i] == '\n')
+						i++;
+					break;
+				}
 			}
 			if (i > 0)
			{
```